### PR TITLE
Update alpine Docker tag to v3.18 & Update Helm release postgresql to ~12.11.0

### PR DESCRIPTION
**Pull request by xbdmHQ**

### CODE REVIEW

File: docker/Dockerfile.arm64 [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM alpine:3.16 AS builder
+FROM alpine:3.18 AS builder
 RUN apk add --no-cache 'crystal=1.4.1-r0' shards sqlite-static yaml-static yaml-dev libxml2-dev zlib-static openssl-libs-static openssl-dev musl-dev
```
Comment (SamantazFox) [inline docker/Dockerfile.arm64:2]: The crystal version needs to be updated:
https://pkgs.alpinelinux.org/packages?name=crystal&branch=v3.18&arch=aarch64

```suggestion
RUN apk add --no-cache 'crystal=1.8.2-r0' shards sqlite-static yaml-static yaml-dev libxml2-dev zlib-static openssl-libs-static openssl-dev musl-dev
```

Comment (unixfox) [reply docker/Dockerfile.arm64:2]: Do we validate that it works great on crystal 1.8?

File: renovate.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (syeopite) [inline renovate.json:1]: Do we want the [renovate bot?](https://github.com/renovatebot/renovate) 

Getting it to automate updating dependencies (do note that it doesn't support shards yet) might make our lives easier

File: docker/Dockerfile [hunk 3]
```diff
@@ -33,7 +33,7 @@ RUN if [[ "${release}" == 1 ]] ; then \
     fi
```
Comment (syeopite) [inline docker/Dockerfile:33]: The crystal installed in this dockerfile also needs to be updated to the latest version

**Review by syeopite — CHANGES_REQUESTED** [review]

File: renovate.json (hunk 2 above)
Comment (unixfox) [reply renovate.json:1]: no we do not, please remove it @xbdmHQ

File: docker/Dockerfile.arm64 (hunk 1 above)
Comment (SamantazFox) [reply docker/Dockerfile.arm64]: I've been running the test instance with it for quite a while, yes.
Also we already have the 1.8 branch in the CI matrix.

Comment (unixfox) [reply docker/Dockerfile.arm64]: ok cool :)

File: renovate.json (hunk 2 above)
Comment (xbdmHQ) [reply renovate.json:1]: > no we do not, please remove it @xbdmHQ

Ok, done.

File: docker/Dockerfile (hunk 3 above)
Comment (xbdmHQ) [reply docker/Dockerfile:27]: > The crystal installed in this dockerfile also needs to be updated to the latest version

Done

File: docker/Dockerfile [hunk 4]
```diff
@@ -31,9 +25,7 @@ RUN if [[ "${release}" == 1 ]] ; then \
         --static --warnings all \
         --link-flags "-lxml2 -llzma"; \
     fi
-
-
-FROM alpine:3.16
+FROM alpine:3.18
```
Comment (unixfox) [inline docker/Dockerfile:28]: Can you keep the double return to the line here?

```
[....]
        --link-flags "-lxml2 -llzma"; \
    fi


FROM alpine:3.18
[....]
```

**Review by unixfox — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 4 above)
Comment (xbdmHQ) [reply docker/Dockerfile:29]: > Can you keep the double return to the line here?
> 
> ```
> [....]
>         --link-flags "-lxml2 -llzma"; \
>     fi
> 
> 
> FROM alpine:3.18
> [....]
> ```

Fixed!

**Comment by unixfox** [comment]
Github actions CI is failing:
```
#26 8.795 Checking player dependencies, this may take more than 20 minutes... If it is stuck, check your internet connection.
#26 279.1 /usr/lib/gcc/aarch64-alpine-linux-musl/12.2.1/../../../../aarch64-alpine-linux-musl/bin/ld: cannot find -lxml2 (this usually means you need to install the development package for libxml2): No such file or directory
#26 279.1 /usr/lib/gcc/aarch64-alpine-linux-musl/12.2.1/../../../../aarch64-alpine-linux-musl/bin/ld: cannot find -llzma (this usually means you need to install the development package for liblzma): No such file or directory
#26 279.8 collect2: error: ld returned 1 exit status
#26 279.8 Error: execution of command failed with exit status 1: cc "${@}" -o /root/.cache/crystal/invidious-scripts-fetch-player-dependencies.cr/macro_run -lxml2 -llzma -rdynamic -static -L/usr/bin/../lib/crystal -lyaml -lz `command -v pkg-config > /dev/null && pkg-config --libs --silence-errors libssl || printf %s '-lssl -lcrypto'` `command -v pkg-config > /dev/null && pkg-config --libs --silence-errors libcrypto || printf %s '-lcrypto'` -lpcre2-8 -lgc -levent
#26 ERROR: process "/bin/sh -c if [[ \"${release}\" == 1 ]] ; then         crystal build ./src/invidious.cr         --release         --static --warnings all         --link-flags \"-lxml2 -llzma\";     else         crystal build ./src/invidious.cr         --static --warnings all         --link-flags \"-lxml2 -llzma\";     fi" did not complete successfully: exit code: 1
------
 > [builder 13/13] RUN if [[ "0" == 1 ]] ; then         crystal build ./src/invidious.cr         --release         --static --warnings all         --link-flags "-lxml2 -llzma";     else         crystal build ./src/invidious.cr         --static --warnings all         --link-flags "-lxml2 -llzma";     fi:
8.795 
8.795 Checking player dependencies, this may take more than 20 minutes... If it is stuck, check your internet connection.
279.1 /usr/lib/gcc/aarch64-alpine-linux-musl/12.2.1/../../../../aarch64-alpine-linux-musl/bin/ld: cannot find -lxml2 (this usually means you need to install the development package for libxml2): No such file or directory
279.1 /usr/lib/gcc/aarch64-alpine-linux-musl/12.2.1/../../../../aarch64-alpine-linux-musl/bin/ld: cannot find -llzma (this usually means you need to install the development package for liblzma): No such file or directory
279.8 collect2: error: ld returned 1 exit status
```

any ideas what is causing that? the correct libraries are being installed here though: https://github.com/iv-org/invidious/pull/4103/files#diff-dbe995fbc7a0984817391ac735bcecd3474f8c4d3aa50361c6af1856bbf7f96fR2

File: docker/Dockerfile [hunk 5]
```diff
@@ -1,26 +1,20 @@
-FROM crystallang/crystal:1.4.1-alpine AS builder
+FROM crystallang/crystal:1.8.2-alpine AS builder
 RUN apk add --no-cache sqlite-static yaml-static
-
```
Comment (SamantazFox) [inline docker/Dockerfile:3]: Why did you remove all these lines?

**Comment by SamantazFox** [comment]
@unixfox Could it be due to `--static`, as you mentioned in an issue or a PR?

File: docker/Dockerfile (hunk 5 above)
Comment (unixfox) [reply docker/Dockerfile:3]: Same in docker/Dockerfile.arm64, there is no need to remove all the existing spaces

**Comment by unixfox** [comment]
Found the issue, we probably now need to use `libxml2-static` instead of `libxml2-dev`.

@xbdmHQ please use this package for Dockerfile.arm64 thank you.

**Comment by xbdmHQ** [comment]
> Found the issue, we probably now need to use `libxml2-static` instead of `libxml2-dev`.
> 
> 
> 
> @xbdmHQ please use this package for Dockerfile.arm64 thank you.

Fixed!

**Comment by unixfox** [comment]
@xbdmHQ please also add `xz-static` package, it seems like it's missing:

```
#25 [builder 12/13] RUN crystal spec --warnings all     --link-flags "-lxml2 -llzma"
#25 20.62 /usr/lib/gcc/aarch64-alpine-linux-musl/12.2.1/../../../../aarch64-alpine-linux-musl/bin/ld: cannot find -llzma (this usually means you need to install the development package for liblzma): No such file or directory
#25 20.65 collect2: error: ld returned 1 exit status
```

**Comment by xbdmHQ** [comment]
> @xbdmHQ please also add `xz-static` package, it seems like it's missing:
> 
> 
> 
> ```
> 
> #25 [builder 12/13] RUN crystal spec --warnings all     --link-flags "-lxml2 -llzma"
> 
> #25 20.62 /usr/lib/gcc/aarch64-alpine-linux-musl/12.2.1/../../../../aarch64-alpine-linux-musl/bin/ld: cannot find -llzma (this usually means you need to install the development package for liblzma): No such file or directory
> 
> #25 20.65 collect2: error: ld returned 1 exit status
> 
> ```

Done

File: docker/Dockerfile (hunk 5 above)
Comment (unixfox) [reply docker/Dockerfile:3]: @xbdmHQ please revert the changes about the spaces removed

**Review by unixfox — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 5 above)
Comment (xbdmHQ) [reply docker/Dockerfile:3]: > @xbdmHQ please revert the changes about the spaces removed

Done

Comment (unixfox) [reply docker/Dockerfile:3]: There are still spaces to modify :smiley:, look at the Dockerfile: https://github.com/iv-org/invidious/pull/4103/files#diff-f34da55ca08f1a30591d8b0b3e885bcc678537b2a9a4aadea4f190806b374ddc

Line 23 removed - please revert
Line 36 unnecessary additional space - please revert

Note: I'm fine with line 2 with the additional space it's ok.

**Review by unixfox — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 5 above)
Comment (xbdmHQ) [reply docker/Dockerfile:3]: > There are still spaces to modify 😃, look at the Dockerfile: https://github.com/iv-org/invidious/pull/4103/files#diff-f34da55ca08f1a30591d8b0b3e885bcc678537b2a9a4aadea4f190806b374ddc
> 
> Line 23 removed - please revert Line 36 unnecessary additional space - please revert
> 
> Note: I'm fine with line 2 with the additional space it's ok.

Fixed

File: docker/Dockerfile [hunk 6]
```diff
@@ -20,7 +21,7 @@ COPY ./videojs-dependencies.yml ./videojs-dependencies.yml
 
 RUN crystal spec --warnings all \
     --link-flags "-lxml2 -llzma"
-
+
```
Comment (unixfox) [inline docker/Dockerfile:24]: Can you remove these useless spaces on this line here? There should be no differences here.

**Review by unixfox — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 6 above)
Comment (xbdmHQ) [reply docker/Dockerfile]: > Can you remove these useless spaces on this line here? There should be no differences here.

done

**Comment by drizzt** [comment]
This breaks captcha since rsvg-convert is not installed anymore. I'm sending a PR to fix that.
